test/rpc/test_rpc_estimate_fee.py

```diff
@@ -180,18 +185,113 @@ def test_estimate_fee_declare(declare_content):
     )
     signature = _get_signature(tx_hash, PREDEPLOYED_ACCOUNT_PRIVATE_KEY)
 
-    declare_transaction = RpcBroadcastedDeclareTxn(
+    declare_transaction = RpcBroadcastedDeclareTxnV1(
         type=declare_content["type"],
-        max_fee=rpc_felt(declare_content["max_fee"]),
+        max_fee=rpc_felt(0),
+        version=hex(SUPPORTED_RPC_TX_VERSION),
+        signature=[rpc_felt(sig) for sig in signature],
+        nonce=rpc_felt(nonce),
+        contract_class=_rpc_contract_class,
+        sender_address=rpc_felt(PREDEPLOYED_ACCOUNT_ADDRESS),
+    )
+
+    response = rpc_call_background_devnet(
+        "starknet_estimateFee", {"request": [declare_transaction], "block_id": "latest"}
+    )
+
+    common_estimate_response(response)
+
+
+@pytest.mark.usefixtures("devnet_with_account")
+def test_estimate_fee_declare_v2():
+    """
+    Test estimating fee for v2 declare transaction.
+    """
+    contract_class, _, compiled_class_hash = load_cairo1_contract()
+
+    version = 2
```

Comment:
Can you extract this `2` to be a constant in the same test file as `SUPPORTED_RPC_TX_VERSION`. It could be something like `SUPPORTED_RPC_DECLARE_TX_VERSION` (so we might also consider having a `DEPRECATED_RPC_DECLARE_TX_VERSION`)

Reply:
Addressed in [fddeb2d](https://github.com/0xSpaceShard/starknet-devnet/pull/467/commits/fddeb2d507a57e9a78e27f681c8539361efeade8)